x/crosschain/keeper/msg_server_vote_inbound_tx.go

```diff
@@ -98,12 +99,16 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 		return nil, err
 	}
 
-	_, isFinalized := k.zetaObserverKeeper.CheckIfFinalizingVote(ctx, ballot)
-	if !isFinalized {
+	_, isFinalizedInThisBlock := k.zetaObserverKeeper.CheckIfFinalizingVote(ctx, ballot)
+	if !isFinalizedInThisBlock {
 		// Return nil here to add vote to ballot and commit state
 		return &types.MsgVoteOnObservedInboundTxResponse{}, nil
 	}
 
+	if k.IsFinalizedInbound(ctx, msg.InTxHash, msg.SenderChainId, msg.EventIndex) {
```

Comment:
It would be better to place this check on an earlier line (e.g. before line 89 `k.zetaObserverKeeper.FindBallot`), as we don't expect observers resubmit message (with modified fields) to create garbage (new) ballots.

Reply:
We still need to allow observers to add votes to ballot even after the tx is finalized

Reply:
Maybe I mean something like this
![image](https://github.com/zeta-chain/node/assets/34498985/94cf6962-9ba0-45f7-b371-63498c045934)

Reply:
Yeah, 
I am. I was just saying that this line needs to be executed before the isFinalizedInbound check. 
https://github.com/zeta-chain/zeta-node/blob/a79b057e6f84ef9166ff5b826aebc3267ba0802f/x/crosschain/keeper/msg_server_vote_inbound_tx.go#L97-L100

Reply:
Added a test 
https://github.com/zeta-chain/zeta-node/blob/5547674b6ba79781ff6e479b7671de74b2202050/x/crosschain/keeper/msg_server_vote_inbound_tx_test.go#L149-L150